### PR TITLE
[3/n] turbo-tasks-fetch

### PR DESCRIPTION
**Pull request by wbinnssmith**
This implements an initial, basic version of retrieving resources (GET-only for now) over http via turbo-tasks. It currently supports specifying the resource url and an optional User-Agent header string, which is needed for upcoming `@next/font` support.

For now, the results of these requests will never be invalidated in the turbo-tasks cache, which is acceptable as the lifetime of the cache is the lifetime of the turbopack process. In the future, we should probably leverage cache context for http like Cache-Control headers to invalidate cache.

Test Plan: Added several unit tests, including that the result is never invalidated and requests are never re-sent.

Closes WEB-190

### CODE REVIEW

**Comment by wbinnssmith** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2769** <a href="https://app.graphite.dev/github/pr/vercel/turbo/2769" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/2769?utm_source=stack-comment).

**Review by jridgewell — APPROVED** [review]

**Comment by wbinnssmith** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/2769?utm_source=stack-comment).


Job Reference: GTwTicpFoR97BYmAqnX0

**Pull request merged**
